tests/test_database.py

```
@@ -0,0 +1,571 @@
+class TestDatabaseManagerErrorHandling:
+    """Additional tests for DatabaseManager error handling and edge cases."""
+
+    @pytest.fixture
+    def db_config(self) -> DatabaseConfig:
+        """Create test database configuration."""
+        return DatabaseConfig(
+            host="localhost",
+            port=5432,
+            name="test_db",
+            user="test_user",
+            password="test_pass",  # pragma: allowlist secret
```

Ruff (0.14.5)

Possible hardcoded password assigned to argument: "password"

(S106)